### PR TITLE
Remove BagReader from public interface

### LINKED ISSUE #27: Consolidate Bag and BagReader classes
I'm not seeing a reason why someone would use BagReader directly. At the least we should simplify the creation of a `Bag` instance and only require a FileLike instance rather than a custom bag reader.

### PR DESCRIPTION
**Pull request by defunctzombie**
Since we already have a FileLike interface to abstract reading bytes in
various environments, BagReader feels like an unecessary added abstraction.

Fixes #27

This change removes it from the public interface so Bag construction only
requires passing in a FileLike.

### CODE REVIEW

File: src/node/index.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+/** @module @foxglove/rosbag/node */
+
+import FileReader from "./FileReader";
+
+export { FileReader };
```
Comment (jtbandes) [inline src/node/index.ts:5]: `export { default as FileReader } from …`

**Comment by defunctzombie** [comment]
@gkjohnson Sorry about that. I think this was caught between refactor work with some remaining items for a new major version and another issue that resulted in making a patch release.

**Comment by defunctzombie** [comment]
> Also I assume this is the only change I need to accommodate the API update? Does anything else need to change?

I think thats the only change. It matches what we are doing in studio https://github.com/foxglove/studio/blob/dcf862f688ce76c7747404b4956d58695c66ee9a/desktop/quicklook/getInfo.ts#L28 so I think you'll be good with that change.

**Comment by gkjohnson** [comment]
Thanks! Figured something like that may have happened -- and thanks confirming my fix!

**Pull request merged**